package.json

```diff
@@ -14,7 +14,19 @@
     "email": "hello@jkimbo.com",
     "url": "jkimbo.com"
   },
-  "main": "lib/",
+  "main": "lib/legacy.js",
+  "exports": {
+    ".": {
+      "import": "./lib/index.js",
```

Comment:
hmm, I don't think we produce ESM?

Reply:
`exports` can benefit CJS projects, like eslint-plugin-jest.

There's a difference in module resolution between the legacy and new config system.

In the legacy config system, as we know,
  - A shareable config or plugin is not loaded(`require` or `import`) in eslintrc.js.
  - Rather, a user just specifies the name as a string (e.g. `plugin:jest/recommended` or `plugin: [ 'jest' ]`), and eslint internally `require()` (not `import()`) them.

In the new config system, 
  - A shareable config or plugin is explicitly loaded by a user, directly or transitively, following the native module resolution by nodejs. eslint does not load by itself anymore.
  - A shareable config or plugin can be `require`d or `import`ed, as eslint support both CJS and ESM for eslint.config.js.

Thus, 

```
if ( require()d ){
  the user is either using legacy or new config system. 
} else if( import()ed ) {
  the user is using new config system
}
```

Therefore, by this PR,
- A user who uses the legacy config system can `require('eslint-plugin-jest')`. This is not breaking change.
- A user who uses the new config system in ESM can `import jest from 'eslint-plugin-jest'`. This feels natural. This can be possible by the [conditional export](https://nodejs.org/api/packages.html#conditional-exports).
- A user who uses the new config system in CJS can `require( 'eslint-plugin-jest/new')`.